packages/core/src/Page/index.js

```diff
@@ -347,7 +347,7 @@ class Page {
     const headingStack = [];
     Object.keys(this.navigableHeadings).forEach((key) => {
       const currentHeadingLevel = this.navigableHeadings[key].level;
-      const currentHeadingHTML = `<a class="nav-link py-1" href="#${key}">`
+      const currentHeadingHTML = `<a onclick="closePageNav();" class="nav-link py-1" href="#${key}">`
```

Comment:
we may catch the bubbled event instead (also checking `e.target` and other parameters to make it as restrictive as possible) to hide the overlay. This way it stays decoupled